code/iaas/external-handler/src/main/java/io/cattle/platform/extension/dynamic/impl/ExternalDynamicExtensionHandlerImpl.java

```diff
@@ -53,37 +54,37 @@
                 break;
             }
         }
-
-        List<? extends ExternalHandler> externalHandlers = externalHandlerDao.getExternalHandler(eventName);
+        
+        List<? extends ExternalHandlerData> externalHandlers = externalHandlerDao.getExternalHandlerData(eventName);
+        
         if (externalHandlers.size() == 0) {
             return Collections.emptyList();
         }
 
         List<Object> result = new ArrayList<Object>(externalHandlers.size());
-        for (ExternalHandler handler : externalHandlers) {
+        for (ExternalHandlerData handler : externalHandlers) {
             result.add(toEventHandler(eventName, handler));
         }
 
         return (List<T>) result;
     }
 
-    protected Object toEventHandler(String eventName, ExternalHandler handler) {
-        Integer retries = DataAccessor.fieldInteger(handler, ExternalHandlerConstants.FIELD_RETRIES);
-        Long timeout = DataAccessor.fieldLong(handler, ExternalHandlerConstants.FIELD_TIMEOUT);
-        String priorityName = DataAccessor.fieldString(handler, ExternalHandlerConstants.FIELD_PRIORITY_NAME);
+    protected Object toEventHandler(String eventName, ExternalHandlerData handler) {
         Integer priority = handler.getPriority();
-
+        String handlerName = handler.getName();
+        
         if (priority == null) {
-            priority = PriorityUtils.getPriorityFromString(priorityName);
+            priority = PriorityUtils.getPriorityFromString(handler.getPriorityName());
         }
 
         EventBasedProcessHandler processHandler = new EventBasedProcessHandler(eventService, objectManager, objectProcessManager, objectMetaDataManager);
 
-        processHandler.setEventName(String.format("%s;handler=%s", eventName, handler.getName()));
-        processHandler.setName(handler.getName());
-        processHandler.setPriority(priority);
-        processHandler.setRetry(retries);
-        processHandler.setTimeoutMillis(timeout);
+        processHandler.setEventName(String.format("%s;handler=%s", eventName, handlerName));
+        processHandler.setName(handlerName);
+        processHandler.setPriority(handler.getPriority());
+        processHandler.setRetry(handler.getRetries());
```

Comment:
How are retries and timeoutMillis getting populated if they came from the data field?